### DISCUSSION TITLE
Dataset by pulsar functions

### DISCUSSION THREAD

**Discussion by codelipenghui** [apache/pulsar#18773] [General]
When use pulsar functions, The ending strategy of consume messages is useful in dataset scenes.

Such as:

1. Ending by message produce time.
2. Ending by message id.
3. Ending by user(user support the ending strategy).

Is this feature suitable for pulsar function? 
The reason for this feature is i think pulsar function can support offline not only stream.

**Comment by srkukarni** [comment]
This feature indeed make sense. Is it possible for you to to volunteer to add this functionality?
